mmdet3d/models/backbones/mink_unet.py

```diff
@@ -0,0 +1,300 @@
+    blob/master/examples/minkunet.py>`_  for more details.
+
+    This backbone is the reimplementation of `U-Net: Convolutional Networks
+    for Biomedical Image Segmentation <https://arxiv.org/abs/1505.04597>`_.
+
+    Args:
+        in_channels (int): Number of input feature channels.
+        out_channels (int): Number of output feature channels.
+        planes (str, optional): Planes unet config.
+            Defaults to None.
+        D (int): Dimension which will be applied in.
+            Defaults to 3.
+    """
+
+    DILATIONS = (1, 1, 1, 1, 1, 1, 1, 1)
+    PLANES = (32, 64, 128, 256, 256, 128, 96, 96)
+    INIT_DIM = 32
+    OUT_TENSOR_STRIDE = 1
+
+    arch_settings = {
+        14: {
+            'block': BasicBlock,
+            'layers': (1, 1, 1, 1, 1, 1, 1, 1),
+            'planes': {
+                'A': (32, 64, 128, 256, 128, 128, 96, 96),
+                'B': (32, 64, 128, 256, 128, 128, 128, 128),
+                'C': (32, 64, 128, 256, 192, 192, 128, 128),
+                'D': (32, 64, 128, 256, 384, 384, 384, 384),
+            },
+        },
+        18: {
+            'block': BasicBlock,
+            'layers': (2, 2, 2, 2, 2, 2, 2, 2),
+            'planes': {
+                'A': (32, 64, 128, 256, 128, 128, 96, 96),
+                'B': (32, 64, 128, 256, 128, 128, 128, 128),
+                'D': (32, 64, 128, 256, 384, 384, 384, 384),
+            },
+        },
+        34: {
+            'block': BasicBlock,
+            'layers': (2, 3, 4, 6, 2, 2, 2, 2),
+            'planes': {
+                'A': (32, 64, 128, 256, 256, 128, 64, 64),
+                'B': (32, 64, 128, 256, 256, 128, 64, 32),
+                'C': (32, 64, 128, 256, 256, 128, 96, 96),
+            },
+        },
+        50: {
+            'block': Bottleneck,
+            'layers': (2, 3, 4, 6, 2, 2, 2, 2),
+        },
+        101: {
+            'block': Bottleneck,
+            'layers': (2, 3, 4, 23, 2, 2, 2, 2),
+        },
+    }
+
+    # To use the model, must call initialize_coords before forward pass.
+    # Once data is processed, call clear to reset the model before calling
+    # initialize_coords
+    def __init__(self, depth, in_channels, planes=None, D=3):
+        super(MinkUNetBase, self).__init__()
+        self.in_channels = in_channels
+        if depth not in self.arch_settings:
+            raise KeyError(f'invalid depth {depth} for unet.')
+
+        if 'planes' not in self.arch_settings[
+                depth] or planes not in self.arch_settings[depth]['planes']:
+            self.planes = self.PLANES
+        else:
+            self.planes = self.arch_settings[depth]['planes'][planes]
+
+        self.block = self.arch_settings[depth]['block']
+        self.layers = self.arch_settings[depth]['layers']
+        self.D = D
+
+        self.network_initialization(self.in_channels, D)
+        self.weight_initialization()
+
+    def weight_initialization(self):
```

Comment:
Need a brief docstring, such as a sentence to describe this function
The coverage rate of docstring is computated in our CI, lower than 95% will make linter fail

Reply:
Please also supplement docstrings for other functions